### DISCUSSION TITLE
Snapshot on motion in ring-client-api

### DISCUSSION THREAD

**Discussion by bailey5** [dgreif/ring#807] [Q&A] [answered]
Hi there,
I'm currently trying to retrieve a snapshot of an event when it occurs. On the data that returns from onNewDing snapshot_url is blank and no other returned fields look like snapshot data.
![](https://i.imgur.com/BEUhwRn.png)
Currently, I'm using getSnapshot() separately to get the snapshot, but most times it takes too long and misses the event.

The Ring app itself returns a snapshot of a motion event in the notification in real-time, so I'm assuming it's something that would be returned in a motion event.

Is this a bug?

Kind Regards,
bailey5

EDIT:
Just saw discussion 657. Seems to be a limitation of battery-powered doorbells. But if that's the case why does the ring app show an instant snapshot?

**Comment by dgreif** [comment, marked as answer]
All of your investigation is accurate.  The Ring app uses a special notification path that is restricted and only works on certain platforms.  Ring actually sends the info to your phone through a special notification URL, but the snapshot _is not_ included in the ding that we can get from the API.  I've asked Ring if we can get access, but so far the answer has been no.  Unfortunately your current setup is the best that's available at this point 😞

**Reply by bailey5** [reply]
Hi dgreif, thanks for your answer. For wired doorbells do they not use that restricted notification path and retrieve the snapshot in the ding data, or is it just that for wired doorbells getSnapshot() works instantly?

**Reply by dgreif** [reply]
Good questions.  It's explained in detail in the wiki: https://github.com/dgreif/ring/wiki/Snapshot-Limitations
